Ginger/Ginger/RunSetLib/RunSetOperations.cs

```diff
@@ -19,23 +19,46 @@ public static RunSetConfig CreateNewRunset(string runSetName="", RepositoryFolde
                 if (!InputBoxWindow.GetInputWithValidation(string.Format("Add New {0}", GingerDicser.GetTermResValue(eTermResKey.RunSet)), string.Format("{0} Name:", GingerDicser.GetTermResValue(eTermResKey.RunSet)), ref runSetName, System.IO.Path.GetInvalidPathChars()))
                 {
                     return null;
-                }                    
+                }
+
+                while (CheckAmbiguity(runSetName))
+                {
+                    Reporter.ToUser(eUserMsgKeys.ValueIssue, runSetName + " already exists ! Run Set name should be unique !");
+
+                    if (!InputBoxWindow.GetInputWithValidation(string.Format("Add New {0}", GingerDicser.GetTermResValue(eTermResKey.RunSet)), string.Format("{0} Name:", GingerDicser.GetTermResValue(eTermResKey.RunSet)), ref runSetName, System.IO.Path.GetInvalidPathChars()))
+                    {
+                        return null;
+                    }
+                }
             }
-            
+
             RunSetConfig runSetConfig = new RunSetConfig();
             runSetConfig.Name = runSetName;
             runSetConfig.GingerRunners.Add(new GingerRunner() { Name = "Runner 1" });
 
             if (runSetsFolder == null)
             {
                 WorkSpace.Instance.SolutionRepository.AddRepositoryItem(runSetConfig);
-            }                
+            }
             else
             {
                 runSetsFolder.AddRepositoryItem(runSetConfig);
-            }                
+            }
 
             return runSetConfig;
         }
+
+        private static bool CheckAmbiguity(string runSetName)
```

Comment:
this method will work but you also could done it with simple Linq like:
if (WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<RunSetConfig>().Where(x=>x.Name.ToLower()==NewName.ToLower()).FirstorNull() != null)